### PR TITLE
Add AccessedVariable struct and logic

### PR DESCRIPTION
**Pull request by spencer-lunarg**
for both https://github.com/KhronosGroup/SPIRV-Reflect/pull/224/ and https://github.com/KhronosGroup/SPIRV-Reflect/pull/226 there is a need to get the underlying mapping between an access (ex `OpLoad`) and the `OpVariable` it touched

Currently we just tracked the `uint32_t` in order to sett the `accessed` boolean

This creates the ground work the for `SpvReflectPrvAccessedVariable` to store this logic

### CODE REVIEW

File: spirv_reflect.c [hunk 1]
```diff
@@ -1104,19 +1128,29 @@ static SpvReflectResult ParseFunction(SpvReflectPrvParser* p_parser, SpvReflectP
...
       case SpvOpStore: {
-        CHECKED_READU32(p_parser, p_node->word_offset + 2, p_func->accessed_ptrs[p_func->accessed_ptr_count]);
-        (++p_func->accessed_ptr_count);
+        const uint32_t result_index = p_node->word_offset + 2;
+        CHECKED_READU32(p_parser, result_index, p_func->accessed_variables[p_func->accessed_variable_count].variable_ptr);
+        (++p_func->accessed_variable_count);
       } break;
       case SpvOpCopyMemory:
       case SpvOpCopyMemorySized: {
-        CHECKED_READU32(p_parser, p_node->word_offset + 2, p_func->accessed_ptrs[p_func->accessed_ptr_count]);
```
Comment (spencer-lunarg) [inline spirv_reflect.c:1116]: added a test for `OpCopyMemory`.. we were off by 1 as the Target and Source operand are `1` and `2`

File: spirv_reflect.c [hunk 2]
```diff
@@ -3046,60 +3080,58 @@ static SpvReflectResult ParseStaticallyUsedResources(SpvReflectPrvParser* p_pars
...
-  for (uint32_t j = 0; j < p_module->descriptor_binding_count; ++j) {
-    SpvReflectDescriptorBinding* p_binding = &p_module->descriptor_bindings[j];
-    bool found = SearchSortedUint32(used_variables, used_variable_count, p_binding->spirv_id);
-    if (found) {
-      p_binding->accessed = 1;
+  for (uint32_t i = 0; i < p_module->descriptor_binding_count; ++i) {
+    SpvReflectDescriptorBinding* p_binding = &p_module->descriptor_bindings[i];
+    for (uint32_t j = 0; j < used_acessed_count; j++) {
+      if (used_accesses[j].variable_ptr == p_binding->spirv_id) {
+        p_binding->accessed = 1;
```
Comment (spencer-lunarg) [inline spirv_reflect.c:3129]: the goal here is we can find all the accesses pointing to this descriptor binding variable

File: spirv_reflect.c [hunk 3]
```diff
@@ -270,23 +290,24 @@ static bool SearchSortedUint32(const uint32_t* arr, size_t size, uint32_t target
   return false;
 }
 
-static SpvReflectResult IntersectSortedUint32(const uint32_t* p_arr0, size_t arr0_size, const uint32_t* p_arr1, size_t arr1_size,
-                                              uint32_t** pp_res, size_t* res_size) {
+static SpvReflectResult IntersectSortedAccessedVariable(const SpvReflectPrvAccessedVariable* p_arr0, size_t arr0_size,
```
Comment (chaoticbob) [inline spirv_reflect.c:293]: This should be one param per line.

File: spirv_reflect.c [hunk 4]
```diff
@@ -270,23 +290,24 @@ static bool SearchSortedUint32(const uint32_t* arr, size_t size, uint32_t target
...
+                                                        const uint32_t* p_arr1, size_t arr1_size, uint32_t** pp_res,
+                                                        size_t* res_size) {
   *pp_res = NULL;
   *res_size = 0;
   if (IsNull(p_arr0) || IsNull(p_arr1)) {
     return SPV_REFLECT_RESULT_SUCCESS;
   }
 
-  const uint32_t* arr0_end = p_arr0 + arr0_size;
+  const SpvReflectPrvAccessedVariable* arr0_end = p_arr0 + arr0_size;
```
Comment (chaoticbob) [inline spirv_reflect.c:302]: Pointer vars should have `p_` prefix.

File: spirv_reflect.c [hunk 5]
```diff
@@ -270,23 +290,24 @@ static bool SearchSortedUint32(const uint32_t* arr, size_t size, uint32_t target
...
   const uint32_t* arr1_end = p_arr1 + arr1_size;
 
-  const uint32_t* idx0 = p_arr0;
+  const SpvReflectPrvAccessedVariable* idx0 = p_arr0;
```
Comment (chaoticbob) [inline spirv_reflect.c:305]: Pointer vars should have `p_` prefix.

File: spirv_reflect.c [hunk 6]
```diff
@@ -3046,60 +3080,58 @@ static SpvReflectResult ParseStaticallyUsedResources(SpvReflectPrvParser* p_pars
...
       ++j;
     }
-    used_variable_count += p_parser->functions[j].accessed_ptr_count;
+    used_acessed_count += p_parser->functions[j].accessed_variable_count;
   }
-  uint32_t* used_variables = NULL;
-  if (used_variable_count > 0) {
-    used_variables = (uint32_t*)calloc(used_variable_count, sizeof(*used_variables));
-    if (IsNull(used_variables)) {
+  SpvReflectPrvAccessedVariable* used_accesses = NULL;
```
Comment (chaoticbob) [inline spirv_reflect.c:3092]: Pointer vars should have `p_` prefix.

**Review by chaoticbob — APPROVED** [review]
Looks good overall. Just some nits with formatting and var names. Not crucial but would be cool to have those changes before merge.

File: spirv_reflect.c (hunk 3 above)
Comment (spencer-lunarg) [reply spirv_reflect.c:293]: https://stackoverflow.com/questions/29483626/clang-format-force-arguments-and-parameters-to-one-line-each-if-the-statement-o

we would need to set `BinPackParameters` in the `.clang-format` file

**Pull request merged**